### PR TITLE
Fix order on new node

### PR DESCRIPTION
**Pull request by Thiblef**
https://github.com/open-orchestra/open-orchestra-model-interface/pull/144
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/503

### CODE REVIEW

**Comment by itkg-ngilain** [comment]
:+1: when tests are ok

File: BackofficeBundle/Tests/Manager/NodeManagerTest.php [hunk 1]
```diff
@@ -288,6 +288,7 @@ public function testInitializeNewNode(NodeInterface $parentNode = null, $status
         $this->assertEquals('fakeNameTheme', $node->getTheme());
         $this->assertEquals('fake keyword', $node->getMetaKeywords());
         $this->assertEquals('fake description', $node->getMetaDescription());
+        $this->assertEquals(0, $node->getOrder());
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Manager/NodeManagerTest.php]: can you add a test where findByParentSortedByOrder return nodes ?

**Comment by Thiblef** [comment]
@nicolasThal @itkg-ngilain @itkg-alavieille  PR updated.

File: BackofficeBundle/Tests/Manager/NodeManagerTest.php [hunk 2]
```diff
@@ -321,6 +322,42 @@ public function provideParentNode()
...
 
     /**
+     * Test getNewNodeOrder
+     *
+     * @param array $nodes
+     * @param int   $count
+     *
+     * @dataProvider provideNodeSortedChildren
+     */
+    public function testGetNewNodeOrder($nodes, $count)
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Manager/NodeManagerTest.php]: add array in signature testGetNewNodeOrder(array $nodes, $count)

File: BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace OpenOrchestra\ModelBundle\Tests\Validator\Constraints;
+
+use Phake;
+use OpenOrchestra\BackofficeBundle\Validator\Constraints\UniqueNodeOrder;
+use OpenOrchestra\BackofficeBundle\Validator\Constraints\UniqueNodeOrderValidator;
+
+/**
+ * Class UniqueNodeOrderValidatorTest
+ * @group tibo
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php]: remove this line

File: BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        $this->assertInstanceOf('Symfony\Component\Validator\ConstraintValidator', $this->validator);
+    }
+
+    /**
+     * @param array $nodes
+     * @param int   $violationTimes
+     *
+     * @dataProvider provideNodesAndCount
+     */
+    public function testAddViolationOrNot($nodes, $violationTimes)
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php]: add array in signature

**Comment by Thiblef** [comment]
@nicolasThal @itkg-ngilain @itkg-alavieille PR updated

File: BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+namespace OpenOrchestra\ModelBundle\Tests\Validator\Constraints;
+
+use Phake;
+use OpenOrchestra\BackofficeBundle\Validator\Constraints\UniqueNodeOrder;
+use OpenOrchestra\BackofficeBundle\Validator\Constraints\UniqueNodeOrderValidator;
+
+/**
+ * Class UniqueNodeOrderValidatorTest
+ */
+class UniqueNodeOrderValidatorTest extends \PHPUnit_Framework_TestCase
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Validator/Constraints/UniqueNodeOrderValidatorTest.php]: change extend by AbstractBaseTestCase

**Comment by nicolasThal** [comment]
:+1: with @itkg-alavieille remark and test running

**Pull request merged**